src/fake-js.ts

```diff
@@ -450,8 +450,14 @@ export function createFakeJsPlugin({
   ): Dep[] {
     const deps = new Set<Dep>()
     const seen = new Set<t.Node>()
+    const inferred = new Set<string>()
 
     walkAST(node, {
+      enter(node) {
+        if (node.type === 'TSInferType' && node.typeParameter) {
+          inferred.add(node.typeParameter.name)
```

Comment:
Should it also remove it from `inferred` when leaving positive branch of `extends`?
Consider 
```ts
type U = string
type Test<T> = T extends Array<infer U> ? U : U
```